src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
+    (m) => m.name === ch.mother
+  ).born);
```

Comment:
Instead of ```forEach```, use the most suitable array method - ```reduce```.
Don’t shorten variables or function names. Check this [guide](https://mate-academy.github.io/style-guides/javascript.html#2311-dont-shorten-variables-or-functions-names).
```suggestion
  const sum = children.reduce(
    (accumulator, child) => (
      accumulator + child.born - people.find(mother => mother.name === child.mother).born
    )
  );
```